### ISSUE TITLE
bug: fastembed 0.4.1 broke our build, please lock it to 0.4.0

### ISSUE THREAD

**Issue by DixitAdh** [NVIDIA/NeMo-Guardrails#820]
### Did you check docs and existing issues?

- [x] I have read all the NeMo-Guardrails docs
- [x] I have updated the package to the latest version before submitting this issue
- [ ] (optional) I have used the develop branch
- [x] I have searched the existing issues of NeMo-Guardrails

### Python version (python --version)

Python 3.10.12

### Operating system/version

Databricks model serving

### NeMo-Guardrails version (if you must use a specific version and not the latest

0.10.0

### Describe the bug

I noticed you are using fastembed>=0.2.2 dependency, they have recently release a new version 0.4.1 in which they have added py-rust-stemmers = "^0.1.0" https://github.com/qdrant/fastembed/commits/main/pyproject.toml this py-rust-stemmers is failing the build with the below error. 
```bash
× Preparing metadata (pyproject.toml) did not run successfully.
#15 139.9   │ exit code: 1
#15 139.9   ╰─> [6 lines of output]
#15 139.9       
#15 139.9       Cargo, the Rust package manager, is not installed or is not on PATH.
#15 139.9       This package requires Rust and Cargo to compile extensions. Install it through
#15 139.9       the system's package manager or via https://rustup.rs/
#15 139.9       
#15 139.9       Checking for Rust toolchain....
#15 139.9       [end of output]
#15 139.9   
#15 139.9   note: This error originates from a subprocess, and is likely not a problem with pip.
#15 139.9 error: metadata-generation-failed
#15 139.9 
#15 139.9 × Encountered error while generating package metadata.
#15 139.9 ╰─> See above for output.
#15 139.9 
#15 139.9 note: This is an issue with the package mentioned above, not pip.
#15 139.9 hint: See above for details.
#15 139.9 
```

To fix this i have to install the fastembed==0.3.6 and lock it in my requirements.txt before installing nemoguardrails

### Steps To Reproduce

https://github.com/qdrant/fastembed/pull/366#issuecomment-2429264651

### Expected Behavior

It is should install fastembed==0.4.0 if you want the latest working version

### Actual Behavior

Build failing due to py-rust-stemmer package

**Comment by Pouyanpi** [comment]
Thank you @DixitAdh for reporting this. We'll do as suggested 👍🏻

**Comment by Pouyanpi** [comment]
Hi @DixitAdh, the most recent version of fastembed is not breaking NeMo Guardrails, so it is compatible. Are you experiencing this incompatibility in another package?

**Comment by Pouyanpi** [comment]
Close It as it is resolved by #884
